### PR TITLE
test serialisation: determine windows extra paths ..

### PR DESCRIPTION
**Pull request by MathieuDuponchelle**
..  for executable arguments too.

This makes it possible to pass an executable to a test, which
can then run it in an appropriate environment.

For context, I encountered this problem when porting harfbuzz:

https://github.com/centricular/harfbuzz/blob/meson/test/fuzzing/meson.build#L23-L29

### CODE REVIEW

**Comment by jpakkane** [comment]
Could we add a test for this?

**Comment by MathieuDuponchelle** [comment]
> Could we add a test for this?

probably yes, I'll look at that tomorrow

**Comment by MathieuDuponchelle** [comment]
Done, let's see what appveyor says :)

File: mesonbuild/backend/backends.py [hunk 1]
```diff
@@ -625,6 +625,8 @@ def write_test_serialisation(self, tests, datafile):
             for a in t.cmd_args:
                 if hasattr(a, 'held_object'):
                     a = a.held_object
+                if isinstance(a, build.Executable):
```
Comment (nirbheek) [inline mesonbuild/backend/backends.py]: I think this should be `build.BuildTarget`, because, f.ex., the argument might be a DLL that is loaded at runtime.

Comment (MathieuDuponchelle) [reply mesonbuild/backend/backends.py]: Done, reckon that should work

**Comment by MathieuDuponchelle** [comment]
Amusing, wrote the test on Linux and it passes everywhere except on Linux oO

**Comment by jpakkane** [comment]
You can download the CI image from Docker hub to run the tests locally.

**Comment by MathieuDuponchelle** [comment]
Turns out the test was failing when cross-compiling. Given this is not what this PR seeks to address, I moved the test to windows. Should pass now (tm).

**Pull request merged**